### PR TITLE
PULL_REQUEST_TEMPLATE.md: Elaborate on testing

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

We've recently seen VM tests being added that could have been simple package tests instead.
The other test types also deserve more attention.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -21,7 +21,11 @@ Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-
...
   - [ ] aarch64-darwin
 - [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
-- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
+- [ ] Tested, as applicable:
+  - [ ] [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
+  - [ ] and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
+  - [ ] or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
+  - [ ] made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
 - [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
 - [ ] Tested execution of all binary files (usually in `./result/bin/`)
```
Comment (Artturin) [inline .github/PULL_REQUEST_TEMPLATE.md:30]: Perhaps this could be improved too. Maybe "tested basic functionality"

Comment (roberth) [reply .github/PULL_REQUEST_TEMPLATE.md]: I think I agree.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -21,9 +21,13 @@ Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-
   - [ ] x86_64-darwin
   - [ ] aarch64-darwin
 - [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
-- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
+- [ ] Tested, as applicable:
+  - [ ] [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
+  - [ ] and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
```
Comment (vcunat) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: Both kinds of tests end up inside `foo.tests` nowadays, right?  So I might be inclined to formulate it that way, though I suspect that in many packages `passthru.tests` still isn't populated.

Comment (vcunat) [reply .github/PULL_REQUEST_TEMPLATE.md:26]: Oh, actually you thought of the linking in the points below.

Comment (vcunat) [reply .github/PULL_REQUEST_TEMPLATE.md:26]: I think I was too hasty with these comments.

**Comment by 06kellyjac** [comment]
Do all of the sub-bullets need checkboxes? I feel like the desire to check all the boxes is what lead to excessive VM tests.
Can the top level have the checkbox and the bullet points explain the appropriate tests for particular circumstances? The downside is that if multiple sub-bullets are required you wouldn't see if they're individually checked off anymore

**Comment by roberth** [comment]
That's all suggestions taken care of. I don't expect any more comments in the short term, so I'll go ahead and merge this. If anyone can think of improvements, I encourage them to open a new PR. :rocket:

**Pull request merged**